cmd/root.go

```diff
@@ -170,20 +170,25 @@ func init() {
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
+	initViper(viper.GetViper())
+}
+func initViper(v *viper.Viper) {
 	if cfgFile != "" { // enable ability to specify config file via flag
-		viper.SetConfigFile(cfgFile)
```

Comment:
So that I can have multiple commands that have a `--role` argument. Otherwise it gets confused, and always uses empty for one of them (the first one bound, I think; binding overrides any existing arguments with the same name).